proto/tendermint/farming/v1beta1/proposal.proto

```diff
@@ -14,10 +15,112 @@ message PublicPlanProposal {
   option (gogoproto.goproto_getters)  = false;
   option (gogoproto.goproto_stringer) = false;
 
-  string title       = 1;
+  // title specifies the title of the plan
+  string title = 1;
+
+  // description specifies the description of the plan
   string description = 2;
 
-  // plans specifies the plan interface(s); it can be FixedAmountPlan or
-  // RatioPlan
-  repeated google.protobuf.Any plans = 3;
+  // name specifies the plan name that describes what plan is for
+  string name = 3;
+
+  // add_request_proposals specifies AddRequestProposal object
+  repeated AddRequestProposal add_request_proposals = 4 [(gogoproto.moretags) = "yaml:\"add_request_proposals\""];
+
+  // update_request_proposals specifies UpdateRequestProposal object
+  repeated UpdateRequestProposal update_request_proposals = 5
+      [(gogoproto.moretags) = "yaml:\"update_request_proposals\""];
+
+  // delete_request_proposals specifies DeleteRequestProposal object
+  repeated DeleteRequestProposal delete_request_proposals = 6
+      [(gogoproto.moretags) = "yaml:\"delete_request_proposals\""];
+}
+
+// AddRequestProposal details a proposal for creating a public plan.
+message AddRequestProposal {
```

Comment:
I think we need to get `Plan.Name` for adding PublicProposal

Reply:
Ok, I will move the `name` field inside the `AddRequestProposal`.

Reply:
How about `UpdateRequestProposal`? Should we allow `name` field to be updatable?

Reply:
I'll add the `name` field in `UpdateRequestProposal` as well.